### DISCUSSION TITLE
use entity outside with(world) {...} block?

### DISCUSSION THREAD

**Discussion by ludek17052** [Quillraven/Fleks#131] [General]
Hello, I just want to ask if I can use entity outside with(world) {...} block?

Now I write:
```
with(world) {
	val playerComponent = entity[PlayerComponent]
}
```

Can I do something like this?:
```
val entity2 = with(world) {
 // ???
}
val playerComponent = entity2[PlayerComponent]
```

**Comment by Quillraven** [comment]
Where are you using that code? In general an entity is just an ID and does not know about its world or components. Therefore it needs to be used in the context of a world to lookup the data of that ID and hence the with(world) syntax.

However if you are in a Fleks system then you don't need it because then you are already in the context of a world.

In short: outside of a Fleks system you need with(world). Otherwise, you can skip it.

**Comment by ludek17052** [comment]
Ok. Thank you.

Where am I using that code?
Sometimes, when the xxxSystem class is long, I copy some function into new class and in this new class I use something like this:

```
    class MyClass(val world: World) {
        fun doSomething() {
            val playerEntity = world.family { all(PlayerComponent) }.first()
            with(world) {
                val playerComponent = playerEntity[PlayerComponent]
		...
            }
        }
        fun doSomethingDifferent() {
            val playerEntity = world.family { all(PlayerComponent) }.first()
            with(world) {
                val playerComponent = playerEntity[PlayerComponent]
		...
            }
        }
    }
```

I do not know if it can be done better, but sometimes I prefer to use more smaller classes than one long class.

**Comment by Quillraven** [comment]
Makes sense. Sometimes you might not need a class and instead you can use extension functions for your world/system. That way you also don't need the with(world).
